File: src/main/java/havarunner/exception/UnsupportedAnnotationException.java
package havarunner.exception;

import java.lang.annotation.Annotation;

public class UnsupportedAnnotationException extends RuntimeException {
    public UnsupportedAnnotationException(Class<? extends Annotation> annotation, Object annotationUser) {
        super(String.format(
            "%s uses the unsupported annotation %s",
            annotationUser,
            annotation
        ));
    }

    public UnsupportedAnnotationException(Annotation annotation, Object annotationUser) {
        this(annotation.getClass(), annotationUser);
    }
}

File: src/main/scala/havarunner/exception/UnsupportedAnnotationException.scala
package havarunner.exception

import java.lang.annotation.Annotation

class UnsupportedAnnotationException(val annotationClass: Class[_ <: Annotation], val annotationUser: Any)
  extends RuntimeException(
    String.format(
      "%s uses the unsupported annotation %s",
      annotationUser.toString,
      annotationClass.getName
    )
  )
